### PR TITLE
[chore] Run lint and test-race GitHub jobs on Windows

### LINKED ISSUE #450: Run go test on Windows
We should also run `go test` on Windows in CI.

We could try running `test-race` job on both Ubuntu and Windows GitHub runners.

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/448#pullrequestreview-1764738324_

### PR DESCRIPTION
**Pull request by scorpionknifes**
Resolves #450
Resloves https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/451

### CODE REVIEW

File: chloggen/internal/chlog/summary.go [hunk 1]
```diff
@@ -75,5 +76,7 @@ func (s summary) String() (string, error) {
 		return "", fmt.Errorf("failed executing template: %w", err)
 	}
 
-	return buf.String(), nil
+	// Replace all \r\n with \n to ensure consistent output across platforms.
+	str := strings.ReplaceAll(buf.String(), "\r\n", "\n")
+	return str, nil
```
Comment (scorpionknifes) [inline chloggen/internal/chlog/summary.go:81]: GitHub Windows Runner pulls the repo with `core.autocrlf=true`, converting the template to `\r\n` line ending.
This results to the test failing
```
expected: "\r\n## 1.0\r\n\r\n### 🛑 Breaking changes 🛑\r\n\r\n- `foo`: broke foo (#123)\r\n- `bar`: broke bar (#345, #678)\r\n  more details\r\n\r\n### 🚩 Deprecations 🚩\r\n\r\n- `foo`: deprecate foo (#1234)\r\n- `bar`: deprecate bar (#3456, #6789)\r\n  more details\r\n\r\n### 🚀 New components 🚀\r\n\r\n- `foo`: new foo (#2)\r\n- `bar`: new bar (#4, #7)\r\n  more details\r\n\r\n### 💡 Enhancements 💡\r\n\r\n- `foo`: enhance foo (#12)\r\n- `bar`: enhance bar (#34, #67)\r\n  more details\r\n\r\n### 🧰 Bug fixe s 🧰\r\n\r\n- `foo`: bug foo (#1)\r\n- `bar`: bug bar (#3, #6)\r\n  more details\r\n"
actual  : "\r\n## 1.0\r\n\r\n### 🛑 Breaking changes 🛑\r\n\r\n- `foo`: broke foo (#123)\r\n- `bar`: broke bar (#345, #678)\n  more details\r\n\r\n### 🚩 Deprecations 🚩\r\n\r\n- `foo`: deprecate foo (#1234)\r\n- `bar`: deprecate bar (#3456, #6789)\n  more details\r\n\r\n### 🚀 New components 🚀\r\n\r\n- `foo`: new foo (#2)\r\n- `bar`: new bar (#4, #7)\n  more details\r\n\r\n### 💡 Enhancements 💡\r\n\r\n- `foo`: enhance foo (#12)\r\n- `bar`: enhance bar (#34, #67)\n  more details\r\n\r\n### 🧰 Bug fixes 🧰\r\n  \r\n- `foo`: bug foo (#1)\r\n- `bar`: bug bar (#3, #6)\n  more details\r\n"
```
Here we clean the output by replacing all `\r\n` to `\n`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -39,14 +39,20 @@ jobs:
         path: ~/.tools
         key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('./internal/tools/**') }}
     - name: Run linters
-      run: make multimod-verify dependabot-check license-check lint
+      run: |
+        make multimod-verify dependabot-check license-check lint
+        echo "Running golangci-lint against go:build windows"
+        GOOS=windows make golangci-lint
```
Comment (scorpionknifes) [inline .github/workflows/ci.yml:45]: https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/451#issuecomment-1842714125

Tested, works as expected

File: chloggen/internal/chlog/summary.go (hunk 1 above)
Comment (scorpionknifes) [reply chloggen/internal/chlog/summary.go:81]: 😢 seems like more tests are failing due to this

Comment (pellared) [reply chloggen/internal/chlog/summary.go:81]: Try adding https://github.com/open-telemetry/opentelemetry-go/blob/main/.gitattributes

**Comment by scorpionknifes** [comment]
Looks like `multimod diff` isn't working on Windows, always thought it was an issue with my local git
```
multimod diff -m tools -p 1.2.3
error running diff: VerifyWorkingTreeClean failed: working tree not clean
```
Related issue with go-git https://www.github.com/go-git/go-git/issues/691 :eyes:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -39,14 +39,20 @@ jobs:
         path: ~/.tools
         key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('./internal/tools/**') }}
     - name: Run linters
-      run: make multimod-verify dependabot-check license-check lint
+      run: |
+        make multimod-verify dependabot-check license-check lint
+        echo "Running golangci-lint against go:build windows"
```
Comment (pellared) [inline .github/workflows/ci.yml:44]:
```suggestion
```

**Review by pellared — DISMISSED** [review]

**Comment by pellared** [comment]
I suggest adding something like

```go
if runtime.GOOS == "windows" {
  t.Skip("diff does not work on Windows. See: https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/467.")
}

to `TestHasChanged` unless you find a quick way to solve https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/467

**Review by pellared — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -39,14 +39,19 @@ jobs:
         path: ~/.tools
         key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('./internal/tools/**') }}
     - name: Run linters
-      run: make multimod-verify dependabot-check license-check lint
+      run: |
+        make multimod-verify dependabot-check license-check lint
+        GOOS=windows make golangci-lint
```
[the remark this answers is not in the document]
Comment (pellared) [reply .github/workflows/ci.yml:44]: To run against Windows code.
See https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/451

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -44,9 +44,41 @@ jobs:
       run: make build
     - name: Check clean repository
       run: make check-clean-work-tree
-
-  test-race:
+  lint-windows:
```
Comment (pellared) [inline .github/workflows/ci.yml:47]: 1. Why not `runs-on: windows-lastest`?
2. Maybe we could try using `strategy: matrix` in the `lint` job like for `test-race`

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -44,9 +44,41 @@ jobs:
...
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.DEFAULT_GO_VERSION }}
+          check-latest: true
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Setup Environment
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
```
Comment (pellared) [inline .github/workflows/ci.yml:60]: This step is no longer needed anywhere.

See https://github.com/open-telemetry/opentelemetry-go/pull/4617 for reference.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (scorpionknifes) [reply .github/workflows/ci.yml]: Good point, wasn't expecting it to work, looks like it's green ✅

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]

**Pull request merged**